google/cloud/storage/blob.py

```diff
@@ -468,9 +474,21 @@ def generate_signed_url(
             raise ValueError("'version' must be either 'v2' or 'v4'")
 
         quoted_name = _quote(self.name, safe=b"/~")
-        resource = "/{bucket_name}/{quoted_name}".format(
-            bucket_name=self.bucket.name, quoted_name=quoted_name
-        )
+
+        if virtual_hosted_style_endpoint and version == "v4":
+            storage_uri = api_access_endpoint.split("://")
+            api_access_endpoint = (
+                storage_uri[0] + "://" + self.bucket.name + "." + storage_uri[1]
+            )
+            resource = "/{quoted_name}".format(quoted_name=quoted_name)
```

Comment:
This could be simplified to `resource = "/{}".format(quoted_name)`, I believe.

Reply:
Tres use the same `resource = "/{quoted_name}".format(quoted_name=quoted_name)` in his PR to maintain the code standard of file and to uniform as they have already used same pattern before, so i think we may have to take care about code uniform.